src/SLCore.Listeners/Implementation/Http/ICertificateDtoConverter.cs

```diff
@@ -19,18 +19,21 @@
  */
 
 using System.ComponentModel.Composition;
-using System.Threading.Tasks;
-using SonarLint.VisualStudio.SLCore.Core;
-using SonarLint.VisualStudio.SLCore.Listener.Proxy;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using SonarLint.VisualStudio.SLCore.Listener.Http.Model;
 
-namespace SonarLint.VisualStudio.SLCore.Listeners.Implementation;
+namespace SonarLint.VisualStudio.SLCore.Listeners.Implementation.Http;
 
-[Export(typeof(ISLCoreListener))]
+internal interface ICertificateDtoConverter
+{
+    X509Certificate2 Convert(X509CertificateDto dto);
+}
+
+[Export(typeof(ICertificateDtoConverter))]
 [PartCreationPolicy(CreationPolicy.Shared)]
-public class ProxyConfigurationListener : IProxyConfigurationListener
+internal class CertificateDtoConverter : ICertificateDtoConverter
 {
-    public Task<SelectProxiesResponse> SelectProxiesAsync(object parameters)
-    {
-        return Task.FromResult(new SelectProxiesResponse());
-    }
+    public X509Certificate2 Convert(X509CertificateDto dto) =>
+        new(Encoding.UTF8.GetBytes(dto.pem));
```

Comment:
I think we should use Convert.FromBase64String, because the PEM format seems to be encoded in base 64